frontend/class-frontend.php

```diff
@@ -1404,7 +1404,7 @@ public function attachment_redirect() {
 			return false;
 		}
 
-		$url = wp_get_attachment_url( get_queried_object_id() );
+		$url = apply_filters( 'wpseo_attachment_redirect', wp_get_attachment_url( get_queried_object_id() ) );
```

Comment:
Could you rename the filter to `wpseo_attachment_redirect_url` as it better describes what the filter is overriding.